test/test_common.cc

```diff
@@ -28,7 +28,7 @@
 #include <nlohmann/json.hpp>
 
 #include "iceberg/json_internal.h"
-#include "iceberg/test/test_config.h"
+#include "test_config.h"
```

Comment:
The reason why this needed to change is that there is an inconsistency between the source directory layout and the build directory layout when it comes to the CMake configuration.  The test_config.h file is located in the `test` directory in the source tree, but CMake installs it to `iceberg/test` in the build directory.

Meson does not allow you to modify the structure of the build directory like that; generally where you place the `meson.build` file in the source tree will reflect where files from that are output into the build directory.

I think the ideal approach would be to move the tests to `iceberg/test` so that the source, build (and potentially install) trees all have the same layout. If we want to do that however, I think that's best to a separate PR

Reply:
Yea - it would require moving the tests directory,  but as long as we are OK with that I can go ahead and submit a PR

Reply:
See https://github.com/apache/iceberg-cpp/pull/247